webpack.config.js

```diff
@@ -0,0 +1,81 @@
+const webpack = require("webpack");
+const path = require("path");
+const UglifyJsPlugin = require("uglifyjs-webpack-plugin");
+
+const outputPath = path.resolve(__dirname, "lib/");
+const inputPath = path.resolve(__dirname, "js_src/");
+
+
+var babelPlugin = new webpack.DefinePlugin({
+  test: /\.jsx?$/,
+  use: {
+    loader: "babel-loader",
+    options: {
+      presets: ["@babel/preset-env", "@babel/preset-react"]
+    }
+  }
+});
+
+var devPlugins = [
+  babelPlugin
+];
+
+var prodPlugins = [
+  babelPlugin,
+  new webpack.DefinePlugin({
+    optimization: {
+      minimizer: [new UglifyJsPlugin({ sourceMap: true })]
+    },
+  })
+];
+
+/// Helper function that generates the webpack export objects array
+///
+/// Usage:
+/// createExports([
+///   [entryFilename, outputFilename, [isProduction]],
+///   {
+///   ...NORMAL WEBPACK CONFIG OBJECT...
+///   },
+///   ...
+/// ]);
+///
+function createExports(exportMappings) {
+  exportObjects = [];
+  exportMappings.forEach(function(mapping) {
+    if (Array.isArray(mapping)) {
+      entryFilename = mapping[0];
+      outputFilename = mapping[1] || mapping[0];
+      isProduction = mapping[2] || outputFilename.includes('_prod');
+      exportObjects.push(
+        {
+          output: {
+            path: outputPath,
+            filename: outputFilename
+          },
+          entry: path.resolve(inputPath, entryFilename),
+          plugins: isProduction ? prodPlugins : devPlugins,
+          mode: isProduction ? "production" : "development",
+          externals: [{ window: "window" }],
+          devtool: isProduction ? false : 'inline-source-map'
```

Comment:
Looking at [the documentation](https://webpack.js.org/configuration/devtool/#production), it seems like `source-map` might be best for all cases:
```suggestion
          devtool: 'source-map'
```

Advantages over current setup:
- development: source maps are a separate file, so they don't have to be loaded when you're not actively using them
- prod: source maps are available for prod builds, allowing prod stack traces to be mapped properly and allowing for easier prod debugging

Reply:
nice